### DISCUSSION TITLE
Convert java to math.net

### DISCUSSION THREAD

**Discussion by ghost** [mathnet/mathnet-numerics#915] [Q&A] [answered]
I'm trying to convert some java code to C# and would like to translate this java code:

`RealMatrix outputmatrix = new SingularValueDecomposition(inputmatrix).getSolver().getInverse();`

How can this be done with math.net numerics?

**Comment by FoggyFinder** [comment]
See

https://numerics.mathdotnet.com/api/MathNet.Numerics.LinearAlgebra.Factorization/Svd%601.htm#Solve

**Reply by ghost** [reply]
Thank you. I tried that but can't get that to work as an oneliner in C#. As a quick solution I just leave out the svd.

**Comment by cdrnet** [comment, marked as answer]
You might be looking for `inputmatrix.PseudeInverse()` (which using SVD)?

 `inputmatrix.Inverse()` is using the LU decomposition, not SVD. Or you could use inputmatrix.Svd().Solve(...) with the identity matrix.

**Reply by ghost** [reply]
Thank you very much. The `Matrix<T> PseudoInverse()` is exactly what i was looking for.
